lib/ipmi_fru.c

```diff
@@ -600,16 +610,12 @@ int write_fru_area(struct ipmi_intf *intf, struct fru_info *fru, uint8_t id,
 			break;
 		}
 
-		if (rsp->ccode == IPMI_CC_REQ_DATA_INV_LENGTH
-		    || rsp->ccode == IPMI_CC_REQ_DATA_FIELD_EXCEED
-		    || rsp->ccode == IPMI_CC_CANT_RET_NUM_REQ_BYTES) {
-			if (fru->max_write_size > 8) {
-				fru->max_write_size -= 8;
-				lprintf(LOG_INFO,
-					"Retrying FRU write with request size %d",
-					fru->max_write_size);
-				continue;
-			}
+		if (fru_cc_rq2big(rsp->ccode) && fru->max_write_size > 8) {
```

Comment:
What is `8`, by the way?

Reply:
Presumably they're using 8 as the magic number for how many bytes to step down their write if it was rejected -- let's try 8 fewer bytes and try again repeatedly while we can.

I could name it, "FRU_WRITE_STEP"